macOS/DuckDuckGo/Permissions/Model/PermissionModel.swift

```diff
@@ -110,11 +117,16 @@ final class PermissionModel {
             permissions[permission].willReload()
         }
         authorizationQueries = []
+        geolocationWasActivated = false
+        removedPermissions.removeAll()
     }
 
     private func updatePermissions() {
         guard let webView = webView else { return }
         for permissionType in PermissionType.permissionsUpdatedExternally {
+            // Skip permissions that were explicitly removed by the user
+            guard !removedPermissions.contains(permissionType) else { continue }
```

Comment:
Bug: removed permissions never updated (never removed from the `removedPermissions`):
- Request a permission
- Open the permission center, choose Never Allow, then remove the permission
- Request the permission again, allow
- Expected: the permission is activated and displayed on the Tab favicon
- Actual: the active permission is not shown